File: pm_core/src/infrastructure/store/mapper/priority_scheme_mapper.go
package mapper

import (
	"github.com/serp/pm-core/src/core/domain/entity"
	"github.com/serp/pm-core/src/infrastructure/store/model"
)

type PrioritySchemeMapper struct{}

func NewPrioritySchemeMapper() *PrioritySchemeMapper {
	return &PrioritySchemeMapper{}
}

func (m *PrioritySchemeMapper) ToEntity(mdl *model.PrioritySchemeModel) *entity.PrioritySchemeEntity {
	if mdl == nil {
		return nil
	}
	return &entity.PrioritySchemeEntity{
		BaseEntity: entity.BaseEntity{
			ID:        mdl.ID,
			CreatedAt: mdl.CreatedAt.UnixMilli(),
			CreatedBy: mdl.CreatedBy,
			UpdatedAt: mdl.UpdatedAt.UnixMilli(),
			UpdatedBy: mdl.UpdatedBy,
		},
		TenantID:          mdl.TenantID,
		Name:              mdl.Name,
		Description:       mdl.Description,
		DefaultPriorityID: mdl.DefaultPriorityID,
	}
}

func (m *PrioritySchemeMapper) ToModel(e *entity.PrioritySchemeEntity) *model.PrioritySchemeModel {
	if e == nil {
		return nil
	}
	return &model.PrioritySchemeModel{
		BaseModel: model.BaseModel{
			ID:        e.ID,
			CreatedBy: e.CreatedBy,
			UpdatedBy: e.UpdatedBy,
		},
		TenantID:          e.TenantID,
		Name:              e.Name,
		Description:       e.Description,
		DefaultPriorityID: e.DefaultPriorityID,
	}
}

func (m *PrioritySchemeMapper) ToEntities(models []*model.PrioritySchemeModel) []*entity.PrioritySchemeEntity {
	if models == nil {
		return nil
	}
	entities := make([]*entity.PrioritySchemeEntity, 0, len(models))
	for _, mdl := range models {
		if e := m.ToEntity(mdl); e != nil {
			entities = append(entities, e)
		}
	}
	return entities
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/mapper/PrioritySchemeMapper.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.mapper;

import org.springframework.stereotype.Component;
import serp.project.pmcore.core.domain.entity.PrioritySchemeEntity;
import serp.project.pmcore.infrastructure.store.model.PrioritySchemeModel;

import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

@Component
public class PrioritySchemeMapper extends BaseMapper {

    public PrioritySchemeEntity toEntity(PrioritySchemeModel model) {
        if (model == null) { return null; }
        return PrioritySchemeEntity.builder()
                .id(model.getId())
                .tenantId(model.getTenantId())
                .name(model.getName())
                .description(model.getDescription())
                .defaultPriorityId(model.getDefaultPriorityId())
                .createdAt(localDateTimeToLong(model.getCreatedAt()))
                .createdBy(model.getCreatedBy())
                .updatedAt(localDateTimeToLong(model.getUpdatedAt()))
                .updatedBy(model.getUpdatedBy())
                .build();
    }

    public PrioritySchemeModel toModel(PrioritySchemeEntity entity) {
        if (entity == null) { return null; }
        return PrioritySchemeModel.builder()
                .id(entity.getId())
                .tenantId(entity.getTenantId())
                .name(entity.getName())
                .description(entity.getDescription())
                .defaultPriorityId(entity.getDefaultPriorityId())
                .createdAt(longToLocalDateTime(entity.getCreatedAt()))
                .createdBy(entity.getCreatedBy())
                .updatedAt(longToLocalDateTime(entity.getUpdatedAt()))
                .updatedBy(entity.getUpdatedBy())
                .build();
    }

    public List<PrioritySchemeEntity> toEntities(List<PrioritySchemeModel> models) {
        if (models == null) { return Collections.emptyList(); }
        return models.stream().map(this::toEntity).collect(Collectors.toList());
    }
}
